tests/raftstore/transport_simulate.rs

```diff
@@ -215,11 +220,39 @@ impl FilterFactory for Isolate {
     }
 }
 
-// Drop all packages for the store with special region.
+#[derive(Clone, Copy)]
+pub enum Direction {
+    Recv,
+    Send,
+    Both,
+}
+
+impl Direction {
+    pub fn is_recv(&self) -> bool {
+        match *self {
+            Direction::Recv | Direction::Both => true,
+            Direction::Send => false,
+        }
+    }
+
+    pub fn is_send(&self) -> bool {
+        match *self {
+            Direction::Send | Direction::Both => true,
+            Direction::Recv => false,
+        }
+    }
+}
+
+/// Drop specified messages for the store with special region.
+///
+/// If `msg_type` is None, all message will be filtered.
 pub struct FilterRegionPacket {
```

Comment:
I see that for all Filter and its corresponding FilterFactory, they have nearly same structure fields and same initialization function, too redundant. Should we think a better way to simplify these? 

/cc @ngaut @BusyJay